File: Challenge-001-Indexes-of-subarray-sum/python/main.py
#!python3

def SubArraySum (input,target):
    arr_size = len(input)
    i=0
    j=0
    sum=0
    while i<arr_size and j<arr_size:
        if target > sum:
            sum+=input[j]
            j+=1
        elif target < sum:
            sum-=input[i]
            i+=1
        else:
            return [i+1,j]
    return [-1]

arr_input = [1,2,3,4,5,6,7,8,9,10]
target = 100;
output = SubArraySum(arr_input,target)
if output[0] == -1:
    print("[",output[0],"]")
    print("There is no subarray with sum",target,".")
else:
    print("[",output[0],",",output[1],"]")
    print("The sum of elements from",output[0],"to",output[1],"is",target,".")

File: Challenge-001-Indexes-of-subarray-sum/rust/src/main.rs

fn SubArraySum (_input: [i32; 10],_target: i32)->[i32; 2]
{
    let mut _i: i32 = 0;
    let mut _j: i32 = 0;
    let mut _sum: i32 = 0;
    let arr_size = _input.len();
    while _i<arr_size as i32 && _j<arr_size as i32{
        if _target > _sum{
            _sum+=_input[_j as usize];
            _j=_j+1;
        }else if _target < _sum{
            _sum-=_input[_i as usize];
            _i=_i+1;
        }else{
            return [_i+1,_j];
        }
    }
    return [-1,0];
}

fn main()
{
    let _arr_input: [i32; 10] = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
    let _target = 100;
    let mut _output: [i32; 2] = SubArraySum(_arr_input,_target);
    if _output[0]==-1{
        println!("[{}]",_output[0]);
        println!("There is no subarray with sum {}",_target)
    }else{
        println!("[{},{}]",_output[0],_output[1]);
        println!("The sum of elements from {} to {} is {}.",_output[0],_output[1],_target);
    }
}